jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Session.java

```diff
@@ -439,27 +442,23 @@ public void onGoAway(final GoAwayFrame frame)
         while (true)
         {
             CloseState current = closed.get();
-            switch (current)
+            if (current == CloseState.NOT_CLOSED)
```

Comment:
Can't the whole `while (true)` loop be replaces with:
```java

    if (closed.compareAndSet(CloseState.NOT_CLOSED, CloseState.REMOTELY_CLOSED)
    {
        closeFrame = frame;
        notifyClose(this, frame, new DisconnectCallback());
        return;
    }
    if (LOG.isDebugEnable())
        LOG.debug("Ignored {}, alread closed", frame);
    return
```

The only time it loops is if the CAS fails and then it is going to return anyway!

Reply:
Done.